app/gui2/src/components/GraphEditor/widgets/WidgetFunction.vue

```diff
@@ -306,9 +317,12 @@ export const widgetDefinition = defineWidget(WidgetInput.isFunctionCall, {
     // and to resolve the infix call as its own application.
     if (prevFunctionState?.callId === ast.id) return Score.Mismatch
 
+    // We only render the function widget on the top-level AST node.
+    if (prevFunctionState?.prefixCalls.includes(ast.id)) return Score.Mismatch
+
     if (ast instanceof Ast.App || ast instanceof Ast.OprApp) return Score.Perfect
 
-    const info = db.getMethodCallInfo(ast.id)
+    const info = getMethodCallInfoRecursively(ast, db)
     if (prevFunctionState != null && info?.partiallyApplied === true && ast instanceof Ast.Ident) {
       return Score.Mismatch
     }
```

Comment:
The purpose of that condition and `partiallyApplied` property was to avoid rendering the argument lists twice in specific case of calling partially applied function from another node:
```
node1 = some_func arg3=baz
node2 = node1 foo bar
```

Since `node1` inside `node2` expression itself has a `Function` type and presents a list of non-applied arguments, those used to be rendered again. With your changes and the `prefixCalls` test above, this condition might no longer be necessary. If that's true, the whole `partiallyApplied` info could be completely removed as well. That would also save us from calling `getMethodCallInfo` in the `score` function, which is really neat.

Reply:
Thank you for explanation. My testing showed that `partiallyApplied` check is indeed redundant for us. I don’t think we can get rid of `getMethodCallInfo` – otherwise we would not build a function widget in case of `node2 = node1`. Below is behavior after my changes:

When no arguments applied to the second node (`node2 = node1`):

<img width="398" alt="Screenshot 2024-04-12 at 3 54 37 PM" src="https://github.com/enso-org/enso/assets/6566674/e204c7a0-1df8-49d5-b887-8a5335d56013">

When arguments applied to the second node (`node2 = node1 path`):

<img width="432" alt="Screenshot 2024-04-12 at 3 56 49 PM" src="https://github.com/enso-org/enso/assets/6566674/219f9d98-6c29-4863-a800-0118120d6377">

I think it looks correct.